tests/components/application-connector/README.md

```diff
@@ -5,19 +5,28 @@ These are the component tests for Application Connector.
 <!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
 **Table of Contents**
 
-- [Design and architecture](#design-and-architecture)
-- [Mock application](#mock-application)
-- [Building](#building)
-- [Running](#running)
-    - [Deploy a Kyma cluster locally](#deploy-a-kyma-cluster-locally)
-    - [Run the tests](#run-the-tests)
-- [Debugging](#debugging)
-    - [Running locally](#running-locally)
-    - [Running without cleanup](#running-without-cleanup)
+- [Component tests for Application Connector](#component-tests-for-application-connector)
+   - [Application Gateway](#application-gateway)
+      - [Design and architecture](#design-and-architecture)
+      - [Mock application](#mock-application)
+         - [Certificates](#certificates)
+         - [API exposed on port `8080`](#api-exposed-on-port-8080)
+         - [API exposed on port `8090`](#api-exposed-on-port-8090)
+         - [API exposed on port `8091`](#api-exposed-on-port-8091)
+      - [Building](#building)
+      - [Running](#running)
+         - [Deploy a Kyma cluster locally](#deploy-a-kyma-cluster-locally)
+         - [Run the tests](#run-the-tests)
+      - [Debugging](#debugging)
+         - [Running locally](#running-locally)
+         - [Running without cleanup](#running-without-cleanup)
+   - [Application Connectivity Validator](#application-connectivity-validator)
```

Comment:
```suggestion
   - [Application Connectivity Validator](#application-connectivity-validator)
       - [Design and architecture](#design-and-architecture-1)
      - [Building](#building-1)
      - [Running](#running-1)
         - [Deploy a Kyma cluster locally](#deploy-a-kyma-cluster-locally-1)
         - [Run the tests](#run-the-tests-1)
      - [Debugging](#debugging-1)
         - [Running without cleanup](#running-without-cleanup-1)
```

I noticed the TOC was not complete. I realized it's because we used the same sub-section titles for both components. I checked how (and if) GitHub handled it, and it does, adding `-1` at the end of the "newer" section titles, so I updated the TOC, but it's a rather hacky way. 
It would probably be best to split this README into two (three, if you count also upcoming Runtime Agent test) separate ones, one for each component.